pkg/engine/planner/physical/optimizer.go

```diff
@@ -248,19 +277,86 @@ func (r *projectionPushdown) applyProjectionPushdown(node Node, projections []Co
 			}
 		}
 		return changed
+	case *ParseNode:
+		// Only apply the pushdown for Metric queries. Log queries should request all keys
+		if !r.isMetricQuery() || len(ambiguousProjections) == 0 {
+			return false
+		}
+
+		// Found a ParseNode - update its keys
+		existingKeys := make(map[string]bool)
```

Comment:
nit: existingKeys is maybe misleading here since it's a combination of the old existing keys and the new projected keys.

Maybe rename to just `requestedKeys`?